ports/stm/common-hal/busio/UART.c

```diff
@@ -209,8 +216,12 @@ void common_hal_busio_uart_construct(busio_uart_obj_t *self,
 
     // Init buffer for rx and claim pins
     if (self->rx != NULL) {
-        if (!ringbuf_alloc(&self->ringbuf, receiver_buffer_size, true)) {
-            mp_raise_ValueError(translate("UART Buffer allocation error"));
+         if (static_rbuf) {
+            self->ringbuf = *static_rbuf;
```

Comment:
Why `*static_rbuf`? Don't you want to store the pointer?

Reply:
I sometimes get confused with assigning pointers etc, what would you do for it?

Reply:
```suggestion
            self->ringbuf = static_rbuf;
```

Reply:
self->ringbuf has a type of ringbuf_t not *ringbuf_t

Reply:
Hmm, this has got me thinking...maybe I can assign the ringbuf in serial.c instead...means I don't have to pass it into the constructor

Reply:
I moved it out of the constructor...much nicer code, managed to remove changes to ~5 files.

Reply:
Ok, now I understand what was happening and would prefer the original way.

The original `*static_rbuf` version was copying the values in the struct into `ringbuf_t` within the object including the buffer length and pointer.

This is better than the version that sets it before because code outside of common_hal shouldn't access struct members except `.base`. The members of a common_hal struct are up to the implementation and external code shouldn't access them directly because they may not existing in some implementations.

Reply:
Ok, I'll fix that.